### PR TITLE
Add Solution

### PR DESCRIPTION
**Pull request by topeogunleye**
Added:

- Created associations between models.
- Wrote SQL queries with Active Record.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Post < ApplicationRecord
+end
```
Comment (keneogu) [inline app/models/post.rb]: There should be an association between models here. example: 
has_many :posts, foreign_key: 'author_id'.
Please add the remaining associations.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class User < ApplicationRecord
+end
```
Comment (keneogu) [inline app/models/user.rb]: Please set up the associations between models. Check the milestone requirements.

**Review by keneogu — CHANGES_REQUESTED** [review]
Hi @topeogunleye, 

Good job so far!

- All linters are passing
- You are following GitHub flow
- Good commit history
- Good PR description

There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️

Please check the milestone and implement the requirements.

- [ ] Add the like and comment models inside the `app/models/` directory, Add set up the associations between models.
- [ ] The User model should have a method that returns the three most recent posts
- [ ] The Post model should have a method that updates the posts counter for a user.
- [ ] The Post model should have a method that returns the 5 most recent comments for a given post.
- [ ] The Comment model should have a method that updates the comments counter for a post.
- [ ] The Comment model should have a method that updates the comments counter for a post.
- [ ] The ApplicationController should contain the `current_user` method

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi @topeogunleye  🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes for processing data in models. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 
✔️  `User`, `Post`, `Comment` and `Like` models are created properly
✔️  All model associations are implemented correctly
✔️  All models have the required methods
✔️  Correct use of Gitflow 
✔️  No linter errors

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**